app/products/calls/connection/connection.test.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {RTCMonitor, RTCPeer, parseRTCStats} from '@mattermost/calls/lib';
+import {Platform} from 'react-native';
+import InCallManager from 'react-native-incall-manager';
+
+import NetworkManager from '@managers/network_manager';
+
+import {newConnection} from './connection';
+import {WebSocketClient} from './websocket_client';
+
+jest.mock('./websocket_client');
+jest.mock('@mattermost/calls/lib');
+
+describe('newConnection', () => {
+    const mockClient = {
+        getWebSocketUrl: jest.fn(() => 'ws://localhost:8065'),
+        getCallsConfig: jest.fn(() => ({})),
+    };
+
+    const mockRTCStats = {
+        iceStats: {
+            succeeded: [
+                {
+                    id: 'candidatePairA',
+                    timestamp: 45,
+                    priority: 45,
+                    state: 'succeeded',
+                    local: {
+                        candidateType: 'host',
+                        protocol: 'udp',
+                        port: 45000,
+                    },
+                    remote: {
+                        candidateType: 'host',
+                        protocol: 'udp',
+                        port: 8443,
+                    },
+                },
+            ],
+            failed: [],
+        },
+    };
+
+    beforeAll(() => {
+        // eslint-disable-next-line
+        // @ts-ignore
+        global.navigator = {};
+
+        // eslint-disable-next-line
+        // @ts-ignore
+        NetworkManager.getClient = jest.fn(() => mockClient);
+
+        // eslint-disable-next-line
+        // @ts-ignore
+        RTCPeer.mockImplementation(() => {
+            return {
+                getStats: jest.fn(),
+                on: jest.fn(),
+            };
+        });
+
+        // eslint-disable-next-line
+        // @ts-ignore
+        RTCMonitor.mockImplementation(() => {
+            return {
+                on: jest.fn(),
+            };
+        });
+
+        Platform.OS = 'web';
```

Comment:
Does not pass when using ios or android?

Reply:
It's mostly to avoid mocking platform-specific native checks we have inside that function, like dealing with audio devices.